src/Core/IAggregateRepository.cs

```diff
@@ -1,8 +1,41 @@
-﻿namespace CR.AggregateRepository.Core
+﻿// <copyright file="IAggregateRepository.cs" company="Cognisant">
+// Copyright (c) Cognisant. All rights reserved.
+// </copyright>
+
+namespace CR.AggregateRepository.Core
 {
+    using Exceptions;
+
+    /// <summary>
+    /// Represents a method of persisting instances of <see cref="IAggregate"/>, handling both storage and access.
+    /// </summary>
     public interface IAggregateRepository
     {
+        /// <summary>
+        /// Persists any uncommitted events on the <see cref="IAggregate"/> to the underlying storage.
+        /// </summary>
+        /// <param name="aggregateToSave">The <see cref="IAggregate"/> to save.</param>
+        /// <exception cref="AggregateVersionException">
+        /// Thrown if <see cref="IAggregate.Version"/> does not match the current version in the underlying storage.
```

Comment:
That isn't correct is it? The IAggregate.Version when saving should be `UnderlyingStorageVersion` + `NewEvents.Count`